### ISSUE TITLE
Token extraction

### ISSUE THREAD

**Issue by Horndrup** [89jd/hass-roborock#20]
Anyone managed to extract the token?

**Comment by u20p17** [comment]
I just get the following lines:

![error](https://user-images.githubusercontent.com/63282286/148521783-43511128-063c-4411-beed-78af462ce3c3.JPG)

I used the following guide: https://github.com/redphx/roborock-tuya-token

Bluestacks 5.5.10.1001 with OnePlus5
roborock_2.3.32_token.apk
Roborock S6 MaxV (Firmware: 02.62.28)

Has anybody found a new solution for getting the token?

**Comment by andras-tim** [comment]
If you can use a rooted Android env with the **latest official** app, you should...

1. add the vacuum to the app and close that
2. go into the `/data/data/com.roborock.smart/files/rr_cache/` directory with elevated rights
3. use `zgrep 'localKey' *` to getting the necessary info (almost all of these files are zip compressed)

(that was worked w/ my **S7 MaxV** on my rooted, Android 10)

**Comment by Horndrup** [comment]
How do you get to the directory with elevated rights?

Do I need a rooted phone?

**Comment by andras-tim** [comment]
> How do you get to the directory with elevated rights?
> 
> Do I need a rooted phone?

Yes. _(as I know, this is required to access the app's internal files.)_
